django/contrib/staticfiles/management/commands/collectstatic.py

```diff
@@ -259,12 +259,8 @@ def delete_file(self, path, prefixed_path, source_storage):
                         full_path = None
                     # Skip the file if the source file is younger
                     # Avoid sub-second precision (see #14665, #19540)
-                    if (target_last_modified.replace(microsecond=0)
-                            >= source_last_modified.replace(microsecond=0)):
-                        if not ((self.symlink and full_path
-                                 and not os.path.islink(full_path)) or
-                                (not self.symlink and full_path
-                                 and os.path.islink(full_path))):
+                    if target_last_modified.replace(microsecond=0) >= source_last_modified.replace(microsecond=0):
+                        if full_path and not (self.symlink ^ os.path.islink(full_path)):
```

Comment:
What I meant is why these can't be intersected in the same conditionnal:

```python
if (target_last_modified.replace(microsecond=0) >= source_last_modified.replace(microsecond=0) and
        full_path and not (self.symlink ^ os.path.islink(full_path)):
```
What's the point of using a nested `if` if theres not `else` block.